radareorg/cutter
# Add more options to the Analysis widget

**Your checklist for this pull request**


**Detailed description**

This MR adds the remaining options to the Analysis Widget that were proposed by @ITAYC0HEN in https://hackmd.io/@Megabeets/H1jBD3Hxv/edit, which are:

- `anal.in`
- `anal.prelude`
- `anal.ptrdepth`

For the `anal.in` option, I selected the following options only:

- All executable maps (`io.maps.x`)
- All maps (`io.maps`)
- Current map (`io.map`)
- Raw (`raw`)
- Current block (`block`)
- Current mapped section (`bin.section`)
- All mapped sections (`bin.sections`)

However, if you think other options should be there, I can add them too.

**Test plan (required)**

1. Open Cutter and go to Edit -> Preferences - > Analysis
2. Select one or more options in the drop-down "Search boundaries for analysis" and check with the console that the value of `anal.in` changes accordingly
3. Change the value of the option "Pointer depth" and check with the console that the value of `anal.ptrdepth` changes accordingly
4. Change the value of the option "Prelude" and check with the console that the value of `anal.prelude` changes accordingly
5. Close and reopen the dialog, and check that the last selected values for these options are maintained

![anal-options-2](https://user-images.githubusercontent.com/26287552/91216870-eece0f80-e716-11ea-8090-51a521ab8130.gif)

**Closing issues**

Closes #875
